evap/student/templates/student_vote.html

```diff
@@ -312,6 +312,259 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
             function textResultsPublishConfirmationBottomChanged() {
                 $("#text_results_publish_confirmation_top").prop('checked', $("#text_results_publish_confirmation_bottom").prop('checked'));
             }
+
+            function selectByNumberKey(row, num) {
+                if (num === 0) {
+                    // Select "No answer"
+                    num = row.children.length;
+                }
+                const index = num - 1;
+
+                if (!(0 <= index && index < row.children.length)) return;
+
+                const nextElement = row.children[index];
+                // TODO: should there be check that nextElement is of type "input" here?
```

Comment:
If the javascript accidentally clicked on the submit button, that would be really bad.

From my understanding of the current code, it is rather unlikely that this happens. What do you think? Do you have any idea if we could add another assertion or similiar on this ("Button must not POST / submit / leave page" or whatever?)

Reply:
Yeah that would be terrible, I guess the suggested safety guard should be here anyway. To correct myself: It should check for the `type` attribute instead of the tagname, because I believe you can also do `<input type="submit">`.

Currently, this is prevented only in the calling code:

```javascript
if (!insideSubmitRow && current.tagName !== "TEXTAREA") {
  // do the number key thing
}
```

Reply:
I added another class that marks elements that should be selectable by pressing the number keys as a safety guard